lib/lambda_ethereum_consensus/beacon/pending_blocks.ex

```diff
@@ -69,6 +74,12 @@ defmodule LambdaEthereumConsensus.Beacon.PendingBlocks do
     |> then(&{:noreply, &1})
   end
 
+  @impl true
+  def handle_cast({:on_tick, time}, state) do
+    ForkChoice.on_tick(time)
```

Comment:
Why do we need this to pass through here? It doesn't use any of pending blocks primitives or state. Is it not possible to call it directly?

Reply:
This is the only way I found to avoid race conditions. Since `ForkChoice` is no longer a GenServer and we are storing its state in the db, the `on_tick` call from `BeaconChain` may modify the state while  an `on_block` call from `PendingBlocks` is being executed.

Reply:
Alright, I'll allow it for this PR. PendingBlocks will no longer be there in a few PRs though, so we'll have to figure that one out.